entity-management-web/src/main/java/eu/europeana/entitymanagement/web/EntityAdminController.java

```diff
@@ -50,56 +50,54 @@
   private static final Logger LOG = LogManager.getLogger(EntityAdminController.class);
 
   private final EntityRecordService entityRecordService;
+  private final SolrService solrService;
   private final EntityUpdateService entityUpdateService;
   private final EntityManagementConfiguration emConfig;
 
   @Autowired
   public EntityAdminController(
       EntityRecordService entityRecordService,
       EntityUpdateService entityUpdateService,
+      SolrService solrService,
       EntityManagementConfiguration emConfig) {
     this.entityRecordService = entityRecordService;
     this.entityUpdateService = entityUpdateService;
+    this.solrService = solrService;
     this.emConfig = emConfig;
   }
 
-  /**
-   * Method to publish to Enrichment
-   *
-   * @param wskey
-   * @param type type of entity
-   * @param identifier entity id
-   * @param request
-   * @return
-   * @throws HttpException
-   */
-  @ApiOperation(
-      value = "Permanent Deletion of Entity",
-      nickname = "deleteEntity",
-      response = java.lang.Void.class)
+  @ApiOperation(value = "Permanent Deletion of Entity", nickname = "deleteEntity")
   @DeleteMapping(
       value = "/{type}/{identifier}/management",
       produces = {HttpHeaders.CONTENT_TYPE_JSONLD, MediaType.APPLICATION_JSON_VALUE})
   public ResponseEntity<String> deleteEntity(
       @RequestParam(value = CommonApiConstants.PARAM_WSKEY, required = false) String wskey,
       @PathVariable(value = WebEntityConstants.PATH_PARAM_TYPE) String type,
       @PathVariable(value = WebEntityConstants.PATH_PARAM_IDENTIFIER) String identifier,
+      @RequestParam(value = WebEntityConstants.QUERY_PARAM_PROFILE, required = false)
+          String profile,
       HttpServletRequest request)
       throws HttpException, EuropeanaApiException {
     if (emConfig.isAuthEnabled()) {
       verifyWriteAccess(Operations.DELETE, request);
     }
     String entityUri = EntityRecordUtils.buildEntityIdUri(type, identifier);
-    Optional<EntityRecord> entityRecordOptional = entityRecordService.retrieveByEntityId(entityUri);
-    if (entityRecordOptional.isEmpty()) {
+    if (!entityRecordService.existsByEntityId(entityUri)) {
       throw new EntityNotFoundException(entityUri);
     }
-    EntityRecord entityRecord = entityRecordOptional.get();
 
-    LOG.info("Permanently deleting entityId={}", entityRecord.getEntityId());
-    entityUpdateService.scheduleTasks(
-        Collections.singletonList(entityRecord.getEntityId()),
-        ScheduledRemovalType.PERMANENT_DELETION);
+    boolean isSynchronous = WebEntityConstants.PARAM_PROFILE_SYNC.equals(profile);
```

Comment:
the profile is cumulative, profile sync,debug should be supported. We should split by comma and than check is syn is present.

Reply:
Good point!

Code updated.